alonzo/impl/src/Cardano/Ledger/Alonzo/TxWitness.hs

```diff
@@ -134,13 +149,25 @@ pattern Redeemers ::
 pattern Redeemers rs <-
   RedeemersConstr (Memo (RedeemersRaw rs) _)
   where
-    Redeemers rs' = RedeemersConstr . memoBytes $ Rec RedeemersRaw !> mapEncode rs'
+    Redeemers rs' =
+      let enc = encodeFoldableEncoder $ \(ptr, (dats, exs)) ->
+            encodeListLen 4
+              <> toCBORGroup ptr
+              <> toCBOR dats
+              <> toCBOR exs
+       in RedeemersConstr $
+            Memo
+              (RedeemersRaw rs')
```

Comment:
I would like to talk about how we might make this simpler. I am not sure how, but there must be a better way.